src/orquestra/sdk/_base/_dsl.py

```diff
@@ -519,14 +520,39 @@ def _validate_task_not_in_main(self):
             )
             raise InvalidTaskDefinitionError(err)
 
+    def _validate_task_resources(self):
+        if self._resources:
+            if self._resources.cpu is not None:
+                cpu = parse_quantity(self._resources.cpu)
+                cpu_int = cpu.to_integral_value()
+                if cpu_int != cpu and cpu > 1:
+                    err = (
+                        f"function {self._fn_name} has CPU resource quantity larger "
+                        f"then 1, which is not a whole number. Make sure any resource "
+                        f">1 is defined as a whole number."
+                    )
+                    raise InvalidTaskDefinitionError(err)
+            if self._resources.gpu is not None:
+                try:
+                    int(self._resources.gpu)
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**question:** ‏Is `"1.0"` an acceptable GPU resource? This code won't work for it.

Reply:
Added a code to parse 1.0 as proper GPU resource, and anything that is not a whole number as invalid